cfme/utils/bz.py

```diff
@@ -207,6 +209,58 @@ def resolve_blocker(self, blocker, version=None, ignore_bugs=None, force_block_s
                 return bug
         return None
 
+    def set_flags(self, idlist, flags):
+        bz_creds = credentials.get("bugzilla", None)
+
+        # login to bzapi
+        if not bz_creds:
+            raise BugzillaError("No bugzilla creds available")
+        try:
+            self.bugzilla.login(bz_creds.get("username"), bz_creds.get("password"))
+        except BugzillaError:
+            logger.exception("Unable to login to Bugzilla with those creds.")
+            raise
+
+        # set the flags
+        for bug in self.bugzilla.getbugs(idlist):
+            result = bug.updateflags(flags)
+            logger.info("Got %s from updating %s", result, bug)
+
+        # logout
+        self.bugzilla.logout()
+
+    def get_bz_info(self, idlist):
+        bz_creds = credentials.get("bugzilla", None)
+
+        # login to bzapi
+        if not bz_creds:
```

Comment:
Can you create a common method for this, since its used in any method actually contacting bugzilla, and we're not having to do the same logic blocks.

Reply:
Additionally, it seems prime for a contextmanger use, since you're explicitly calling login/logout before/after BZ queries.